worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/consumption/StagingQueueWeightSettingsProvider.java

```diff
@@ -0,0 +1,109 @@
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.Collections;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StagingQueueWeightSettingsProvider {
+    private static final Logger FAST_LANE_LOGGER = LoggerFactory.getLogger("FAST_LANE");
+    ArrayList<String> regexWeightStrings = new ArrayList<>();
+
+    public Map<String, Double> getStagingQueueWeights(List<String> stagingQueueNames) {
+
+        final Map<String, Double> stagingQueueWeights = new HashMap<>();
+
+        final Set<Map.Entry<String, String>> envVariables = EnvVariableCollector.getEnvVariables();
+
+        for (final Map.Entry<String, String> entry : envVariables) {
+
+            // This matches strings with regex followed by comma and number. This is dependent on no whitespace
+            final String regexEnvMatcher = "[^.]*,(?!.*,)(0|[1-9]\\d*)?(\\.\\d+)?(?<=\\d)$";
+
+            Pattern pattern = Pattern.compile(regexEnvMatcher);
+
+            Matcher matcher = pattern.matcher(entry.getValue());
+
+            if (matcher.matches()) {
+                FAST_LANE_LOGGER.debug("Env variable {} matches the format to alter weight of worker", entry.getValue());
+                regexWeightStrings.add(entry.getValue());
+            }
+        }
+
+        final Map<Pattern, Double> regexToWeightMap = new HashMap<>();
+
+        // Loop over the environment variable strings added to get regex and weight
+        for (final String regexWeightString : regexWeightStrings) {
+            try {
+                final String[] regexPattern = regexWeightString.split(",(?!.*,)");
+                regexToWeightMap.put(Pattern.compile(regexPattern[0]), Double.parseDouble(regexPattern[1]));
+            } catch (ArrayIndexOutOfBoundsException e) {
+                FAST_LANE_LOGGER.error("Invalid Regex string, ensure the format is regex pattern followed by a comma, followed by the " +
+                        "weight to be added to the string matching the regex.");
+                throw e;
+            }
+        }
+
+        for (final String stagingQueue : stagingQueueNames) {
+
+            final Map<Integer, Double> matchLengthToWeight = new HashMap<>();
+
+            // Find the regex strings that match the staging queue
+            for (final Map.Entry<Pattern, Double> entry : regexToWeightMap.entrySet()) {
+                Matcher matcher = entry.getKey().matcher(stagingQueue);
+
+                while (matcher.find()) {
+                    int matcherGroupCount = matcher.groupCount();
+                    for (int groupCount = 0; groupCount <= matcherGroupCount; groupCount++) {
+                        // Map the length of the string match to the weight that has been matched
+
+                        // If there are 2 regex patterns that match the same length of string but provide different weights,
```

Comment:
Good decision, can you investigate changing the following "if else" with a call to map.compute